### PR TITLE
Add support AWS Embedded Metric Format Version 0

### PR DESCRIPTION
**Pull request by khanhntd**
Same as this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20314

**Description:** 
Currently, `awsemfexporter` will group all the metrics, translate to fields and setting the fields with exported metrics as EMF version 1 -
```
"_aws": {
					"CloudWatchMetrics": [
					{
						"Namespace": "ECS",
						"Dimensions": [ ["ClusterName"] ],
						"Metrics": [{"Name": "memcached_commands_total"}]
					}
					],
					"Timestamp": 1668387032641
			  	}
```

Therefore, adding support for EMF version v0 and a flag for customers to make a difference between Version 0 and Version 1 (e.g EMF v0)
```
				{
					"Version": 0,
					"CloudWatchMetrics": [
					{
						"Namespace": "ECS",
						"Dimensions": [ ["ClusterName"] ],
						"Metrics": [{"Name": "memcached_commands_total"}]
					}
					],
					"Timestamp": 1668387032641
			  	}
```

This can be validated by send metrics with `awsemfexporter`  

image

**Link to tracking Issue:** <Issue number if applicable>
N/A

**Testing:** <Describe what testing was performed and which tests were added.>
There are two tests I have done:
* By running `go test` with `awsemfexporter`, the result is
```
khanhntd@88665a0eaf54 awsemfexporter % go test ./...
ok      github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter       2.409s
```

* By using the following configuration for `awsemfexporter` with prometheus,  here is the following result: 
image
**Documentation:** <Describe the documentation added.>
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html#CloudWatch_Embedded_Metric_Format_Specification_structure

### CODE REVIEW

File: exporter/awsemfexporter/factory.go [hunk 1]
```diff
@@ -47,18 +49,32 @@ func createDefaultConfig() component.Config {
 		LogStreamName:                   "",
 		Namespace:                       "",
 		DimensionRollupOption:           "ZeroAndSingleDimensionRollup",
+		EnableEMFVersion1:               true,
```
Comment (khanhntd) [inline exporter/awsemfexporter/factory.go:52]: The [default EMF version is v1](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsemfexporter/metric_translator.go#L374-L378) and the public document mentions [V1 more than V0](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html#CloudWatch_Embedded_Metric_Format_Specification_structure). Therefore, enable EMF version with default is true

File: exporter/awsemfexporter/metric_declaration.go [hunk 2]
```diff
@@ -186,8 +186,8 @@ func (lm *LabelMatcher) init() (err error) {
 	if len(lm.Separator) == 0 {
 		lm.Separator = ";"
 	}
-	lm.compiledRegex = regexp.MustCompile(lm.Regex)
-	return
+	lm.compiledRegex, err = regexp.Compile(lm.Regex)
```
Comment (khanhntd) [inline exporter/awsemfexporter/metric_declaration.go:189]: `MustCompile`  will panic if the [regexp is not compilable](https://pkg.go.dev/regexp#MustCompile). Therefore, using `Compile` which are the same with MustCompile but return readable error instead of panic

File: exporter/awsemfexporter/config.go [hunk 3]
```diff
@@ -102,6 +108,8 @@ type MetricDescriptor struct {
 	Overwrite bool `mapstructure:"overwrite"`
 }
 
+var _ component.Config = (*Config)(nil)
```
Comment (khanhntd) [inline exporter/awsemfexporter/config.go:111]: This will validate if the `Config` declared by `awsemfexporter Config` has all the declared function by the [interface](https://stackoverflow.com/a/27804417)

File: exporter/awsemfexporter/emf_exporter.go [hunk 4]
```diff
@@ -56,66 +51,40 @@ type emfExporter struct {
...
 
 	// create CWLogs client with aws session config
-	svcStructuredLog := cwlogs.NewClient(logger, awsConfig, params.BuildInfo, expConfig.LogGroupName, expConfig.LogRetention, session)
-	collectorIdentifier, _ := uuid.NewRandom()
+	svcStructuredLog := cwlogs.NewClient(set.Logger, awsConfig, set.BuildInfo, config.LogGroupName, config.LogRetention, session)
+	collectorIdentifier, err := uuid.NewRandom()
+
+	if err != nil {
+		return nil, err
+	}
```
Comment (chadpatel) [inline exporter/awsemfexporter/emf_exporter.go:74]: why are we bothering to handle this error?

File: exporter/awsemfexporter/emf_exporter.go [hunk 5]
```diff
@@ -56,66 +51,40 @@ type emfExporter struct {
...
 	}
 
-	logger := params.Logger
-	expConfig := config.(*Config)
-	expConfig.logger = logger
+	config.logger = set.Logger
 
 	// create AWS session
-	awsConfig, session, err := awsutil.GetAWSConfigSession(logger, &awsutil.Conn{}, &expConfig.AWSSessionSettings)
+	awsConfig, session, err := awsutil.GetAWSConfigSession(set.Logger, &awsutil.Conn{}, &config.AWSSessionSettings)
```
Comment (chadpatel) [inline exporter/awsemfexporter/emf_exporter.go:63]: why is this changing?  It seems like you are refactoring things that are unrelated to the specific change necessary for EMFv0

There are a few examples of that in this PR

**Review by chadpatel — COMMENTED** [review]
Added a few small comments.  I am mostly wondering why we want to handle the uuid error that is extremely unlikely

File: exporter/awsemfexporter/emf_exporter.go (hunk 4 above)
Comment (williazz) [reply exporter/awsemfexporter/emf_exporter.go:74]: which design principle says to accept unhandled errors because they are unlikely?

File: exporter/awsemfexporter/emf_exporter.go (hunk 5 above)
Comment (williazz) [reply exporter/awsemfexporter/emf_exporter.go:63]: yeah i agree. if you must include unrelated changes in a single PR, then i'd appreciate comments saying that they are unrelated

Comment (chadpatel) [reply exporter/awsemfexporter/emf_exporter.go:63]: IMO - better to refactor in a different PR

File: exporter/awsemfexporter/emf_exporter.go (hunk 4 above)
Comment (chadpatel) [reply exporter/awsemfexporter/emf_exporter.go:74]: I am more trying to get at what changed.  This is one of those type of methods that will never fail unless something is completely broken

File: exporter/awsemfexporter/README.md [hunk 6]
```diff
@@ -32,12 +32,13 @@ The following exporter configuration parameters are supported.
...
...Logs with Embedded Metric Format in selected version [(e.g version 1 with _aws)](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html#CloudWatch_Embedded_Metric_Format_Specification_structure), version 0 without _aws)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | `1`                                                                                        |
```
Comment (chadpatel) [inline exporter/awsemfexporter/README.md:37]: the critical thing to communicate to our customers is what the default is (i.e. 1) and that it is optional.  It is also redundant to call it `emf_version` as we are in the emf exporter.  Maybe output_version? I don't care as much about the naming as the lack of documentation on how this should be used (i.e., don't use it, the default is fine :))

**Review by chadpatel — CHANGES_REQUESTED** [review]

File: exporter/awsemfexporter/README.md (hunk 6 above)
Comment (khanhntd) [reply exporter/awsemfexporter/README.md:37]: I disagree `output_version` would make confusing with the `output_destination`. Would ` output_version=2` means we are exporting `emf version v2` to `cloudwatch` or we are exporting  `emf` to `cloudwatch version 2`?

Agree with you on this
```
the critical thing to communicate to our customers is what the default is (i.e. 1) and that it is optional
```

That's why default exists right? For optional configuration instead of force customers to have expertise in it.

File: exporter/awsemfexporter/emf_exporter.go (hunk 4 above)
Comment (khanhntd) [reply exporter/awsemfexporter/emf_exporter.go:74]: If its failed, it would fail at `unit test`. As billy said, why would we ignore unhandled errors, if it is in test environment ,yes I agree it should be more less robust but I don't think we want to enforce it with prod env.

File: exporter/awsemfexporter/emf_exporter.go (hunk 5 above)
Comment (khanhntd) [reply exporter/awsemfexporter/emf_exporter.go:63]: I'm fine with creating in different PR. However, its only simple line change and there are hardly and complication in it. If you are saying refactoring more than 5 files with more than 50 lines then yes, I would agree with you.

File: exporter/awsemfexporter/README.md (hunk 6 above)
Comment (khanhntd) [reply exporter/awsemfexporter/README.md:37]: Change to `version` for less redundant

**Comment by williazz** [comment]
Did you do any testing?

**Comment by khanhntd** [comment]
> Did you do any testing?

Its in PR description
